src/Asset/Capacity/HasContractsCapacity.php

```diff
@@ -47,6 +48,37 @@ public function getLabel(): string
         return Contract::getTypeName();
     }
 
+    public function canBeDisabled(CommonDBTM $asset, string $table = null): bool
+    {
+        return parent::canBeDisabled($asset, Contract_Item::getTable());
+    }
+
+    public function getCapacityUsageDescription(\CommonDBTM $asset): string
+    {
+        /** @var \DBmysql $DB */
+        global $DB;
+
+        $count = $DB->request([
+            'SELECT'   => 'contracts_id',
+            'DISTINCT' => true,
+            'FROM'     => Contract_Item::getTable(),
+            'WHERE'    => [
+                'itemtype' => 'Glpi\CustomAsset\\' . $asset->fields['system_name'],
+            ],
+        ])->count();
+
+        $max = $DB->request([
+            'SELECT'   => 'items_id',
+            'DISTINCT' => true,
+            'FROM'     => Contract_Item::getTable(),
+            'WHERE'    => [
+                'itemtype' => 'Glpi\CustomAsset\\' . $asset->fields['system_name'],
+            ],
+        ])->count();
```

Comment:
This code seems really redundant. You should probably declare methods in the abstract class, like:
-  `protected funtion countTargetItemUsed(string $source_itemtype, string $target_itemtype): int`
-  `protected funtion countSourceItemCapacityUsages(string $source_itemtype, string $target_itemtype): int`